### PR TITLE
Add TYPO3 12 compatibility + Fix PHP8 issue in middleware

### PR DESCRIPTION
**Pull request by mediaessenz**

### CODE REVIEW

File: ext_localconf.php [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <?php
-defined('TYPO3_MODE') or die();
+defined('TYPO3') or die();
```
Comment (bmack) [inline ext_localconf.php:2]: if we do this, we can only support v10.4+.

What I've done in the past "defined('TYPO3') or defined('TYPO3_MODE') or die();" - can you update that line, please?

Comment (mediaessenz) [reply ext_localconf.php]: Done.

**Comment by bmack** [comment]
we still lack support for `$GLOBALS['TYPO3_CONF_VARS']['SC_OPTIONS']['urlProcessing']['urlProcessors']['jumpurl']['processor']` in TYPO3 v12 I guess, but this is a good start!

**Comment by mediaessenz** [comment]
> we still lack support for `$GLOBALS['TYPO3_CONF_VARS']['SC_OPTIONS']['urlProcessing']['urlProcessors']['jumpurl']['processor']` in TYPO3 v12 I guess, but this is a good start!

See: https://github.com/FriendsOfTYPO3/jumpurl/pull/42

**Pull request merged**